Dockerfile

```diff
@@ -15,14 +17,13 @@ ARG service_alias
 ARG service_controller_git_version
 ARG service_controller_git_commit
 ARG build_date
-ARG go_arch=amd64
 # The directory within the builder container into which we will copy our
 # service controller code.
 ARG work_dir=/github.com/aws-controllers-k8s/$service_alias-controller
 WORKDIR $work_dir
 ENV GOPROXY=https://proxy.golang.org|direct
 ENV GO111MODULE=on
-ENV GOARCH=$go_arch
+ENV GOARCH=$TARGETARCH
```

Comment:
If I understand correctly, this is a variable rename. I'm afraid this might break some of the pipelines that might inject `go_arch` instead of `TARGETARCH`. Are you using this script internally and you rely on this specific variable?

Reply:
https://www.docker.com/blog/faster-multi-platform-builds-dockerfile-cross-compilation-guide/#93cb
The docker buildx will inject the architecture into the TARGETARCH.
This is practically standard and cannot be changed.

Reply:
Now we already support test-infra as well.
Also, my understanding is that the internal pipeline is currently only running on amd64, so it will not break.
https://github.com/aws-controllers-k8s/test-infra/pull/383